documentation/modules/ROOT/pages/connectors/sqlserver.adoc

```diff
@@ -469,6 +476,9 @@ When we compare this to the value in the _insert_ event, we see a couple of diff
 * The `before` field now has the state of the row with the values before the database commit
 * The `after` field now has the updated state of the row, and here was can see that the `email` value is now `noreply@example.org`.
 * The `source` field structure has the same fields as before, but the values are different since this event is from a different position in the transaction log.
+* The `event_serial_no` field has value `2`.
```

Comment:
There should be an explanation of that field's purpose. So far it's not mentioned at all (look for the description of `source` under [change event values](https://debezium.io/documentation/reference/0.10/connectors/sqlserver#change-event-values). In particular I think a description of the specific semantics in case of connector restart after a crash are warranted (AFAIU, the same event may produced again may have a lower serial no, depending on the committed offset).

Reply:
@gunnarmorling Docs added, BTW the serial no is always same even after restart.

Reply:
Thanks, that explanation helped. In fact my interpretation of the new field was a different one than it actually is: i thought it'd count the events within one transaction, whereas actually its scope is the unit of same commit and change LSN.

> the serial no is always same even after restart.

Ok, that's great. Just to be sure: is it ensured that we wouldn't commit the offset after the first of the two events (e.g. after the delete in case of a PK update)? Because if that could happen, then we would restart from the insert event and use serial no = 1. But I suppose it cannot happen, as we'd always return the two events within one and the same call to `poll()` and it seems unlikely that KC would commit offsets in the middle of such a batch of values it received. Correct?

Reply:
@gunnarmorling The restart can happen in the middle - but event_serial_no is part of offsets so this is covered.

Reply:
Ah, ok. That makes sense. Thanks!